services/minespace-web/src/components/Forms/projects/majorMineApplication/MajorMineApplicationForm.tsx

```diff
@@ -0,0 +1,365 @@
+
+import DocumentTable from "@mds/common/components/documents/DocumentTable";
+import RenderField from "@mds/common/components/forms/RenderField";
+import ArchivedDocumentsSection from "@mds/common/components/projects/ArchivedDocumentsSection";
+interface MajorMineApplicationFormProps {
+  project: IProject;
+  handleSubmit: () => void;
+  refreshData: () => void;
+}
+
+const MajorMineApplicationForm: React.FC<MajorMineApplicationFormProps> = ({
+  project,
+  handleSubmit,
+  refreshData,
+}) => {
+  const dispatch = useDispatch();
+
+  const { primary_documents, spatial_documents, supporting_documents } =
+    useSelector(getFormValues(FORM.ADD_MINE_MAJOR_APPLICATION)) || {};
+  const mineDocuments = useSelector(getMineDocuments);
+  const [uploadedFiles, setUploadedFiles] = useState<any[]>([]);
+  const { isFeatureEnabled } = useFeatureFlag();
+
+  const acceptedFileTypesMap = {
+    ...SPATIAL,
+    ...DOCUMENT,
+    ...MODERN_EXCEL,
+  };
+
+  const onFileLoad = (
+    fileName: string,
+    document_manager_guid: string,
+    documentTypeCode: string,
+    documentTypeField: string
+  ) => {
+    const newFiles = [
+      ...uploadedFiles,
+      {
+        document_name: fileName,
+        document_manager_guid,
+        major_mine_application_document_type_code: documentTypeCode,
+        mine_guid: project?.mine_guid,
+      },
+    ];
+    setUploadedFiles(newFiles);
+    dispatch(
+      change(
+        FORM.ADD_MINE_MAJOR_APPLICATION,
+        documentTypeField,
+        newFiles.filter(
+          (file) => file?.major_mine_application_document_type_code === documentTypeCode
+        )
+      )
+    );
+  };
+
+  const onRemoveFile = (
+    err: any,
+    fileItem: any,
+    documentTypeFieldForm: string,
+    documentsForm: any[]
+  ) => {
+    const newFiles = uploadedFiles?.filter(
+      (file) => file.document_manager_guid !== fileItem.serverId
+    );
+
+    setUploadedFiles(newFiles);
+
+    const newDocumentsForm = documentsForm?.filter(
+      (doc) => doc.document_manager_guid !== fileItem.serverId
+    );
+
+    dispatch(change(FORM.ADD_MINE_MAJOR_APPLICATION, documentTypeFieldForm, newDocumentsForm));
+  };
+
+  const uniqueDocs = (formDocuments: any[], mmaDocuments: any[], type_code: string) => {
+    const validFormDocuments = Array.isArray(formDocuments) ? formDocuments : [];
+
+    const documents = [
+      ...validFormDocuments,
+      ...(Array.isArray(mmaDocuments)
+        ? mmaDocuments.filter((doc) => doc.major_mine_application_document_type_code === type_code)
+        : []),
+    ];
+
+    const existingDocumentGuids = new Set<string>();
+    const uniqueDocuments = documents.filter((doc) => {
+      const isDuplicate = existingDocumentGuids.has(doc.document_manager_guid);
+      existingDocumentGuids.add(doc.document_manager_guid);
+      return !isDuplicate;
+    });
+
+    return uniqueDocuments;
+  };
+
+  const openSpatialDocumentModal = () => {
```

Comment:
I might be behind on what is required for this story, but do we need to setup the spatial file bundle process to final applications on the Core side as well?